src/app/CommandHandler.cpp

```diff
@@ -312,6 +314,91 @@ CHIP_ERROR CommandHandler::ProcessCommandDataIB(CommandDataIB::Parser & aCommand
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR CommandHandler::ProcessGroupCommandDataIB(CommandDataIB::Parser & aCommandElement)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    CommandPathIB::Parser commandPath;
+    TLV::TLVReader commandDataReader;
+    ClusterId clusterId;
+    CommandId commandId;
+    GroupId groupId;
+    FabricIndex fabric;
+
+    Credentials::GroupDataProvider::GroupEndpoint mapping;
+    Credentials::GroupDataProvider * groupDataProvider = Credentials::GetGroupDataProvider();
+    Credentials::GroupDataProvider::EndpointIterator * iterator;
+
+    err = aCommandElement.GetPath(&commandPath);
+    SuccessOrExit(err);
+
+    err = commandPath.GetClusterId(&clusterId);
+    SuccessOrExit(err);
+
+    err = commandPath.GetCommandId(&commandId);
+    SuccessOrExit(err);
+
+    groupId = mpExchangeCtx->GetSessionHandle()->AsGroupSession()->GetGroupId();
+    fabric  = GetAccessingFabricIndex();
+
+    ChipLogDetail(DataManagement,
+                  "Received group command for Group=%" PRIu16 " Cluster=" ChipLogFormatMEI " Command=" ChipLogFormatMEI, groupId,
+                  ChipLogValueMEI(clusterId), ChipLogValueMEI(commandId));
+
+    err = aCommandElement.GetData(&commandDataReader);
+    if (CHIP_END_OF_TLV == err)
+    {
+        ChipLogDetail(DataManagement,
+                      "Received command without data for Group=%" PRIu16 " Cluster=" ChipLogFormatMEI " Command=" ChipLogFormatMEI,
+                      groupId, ChipLogValueMEI(clusterId), ChipLogValueMEI(commandId));
+        err = CHIP_NO_ERROR;
+    }
+    SuccessOrExit(err);
+
+    iterator = groupDataProvider->IterateEndpoints(fabric);
+    VerifyOrExit(iterator != nullptr, err = CHIP_ERROR_NO_MEMORY);
+
+    while (iterator->Next(mapping))
+    {
+        if (groupId != mapping.group_id)
+        {
+            continue;
+        }
+
+        ChipLogDetail(DataManagement,
+                      "Processing group command for Endpoint=%" PRIu16 " Cluster=" ChipLogFormatMEI " Command=" ChipLogFormatMEI,
+                      mapping.endpoint_id, ChipLogValueMEI(clusterId), ChipLogValueMEI(commandId));
+
+        const ConcreteCommandPath concretePath(mapping.endpoint_id, clusterId, commandId);
+
+        if (!mpCallback->CommandExists(concretePath))
+        {
+            ChipLogError(DataManagement, "No Cluster " ChipLogFormatMEI " on Endpoint 0x%" PRIx16, ChipLogValueMEI(clusterId),
+                         mapping.endpoint_id);
+
+            continue;
+        }
+
+        if ((err = MatterPreCommandReceivedCallback(concretePath)) == CHIP_NO_ERROR)
```

Comment:
There needs to be an ACL check before this, no?  There is definitely one in the unicast case....

Reply:
ACL isn't done for groups yet. I added a comment with the same code as the unicast case.